### DISCUSSION TITLE
Is there a way to exclude an specific field from export?

### DISCUSSION THREAD

**Discussion by JonathanLoscalzo** [pydantic/pydantic#5608] [Question] [answered]
I need to export a model (JSON or dict). I need to export all but one feature with an specific value or condition

I know the options are `exclude_unset`, `exclude_defaults`, but these options are limited to all fields.

My current requirement is: export a model but if one field has an specific value, it should be excluded.
Is there any way? 

If it does not exist: 
- it could be an option in the Field class 
- it could be a new argument "_exclude_by_" that accepts a _dict[str, Callable[..., bool]]_

**Comment by chrisK824** [comment, marked as answer]
As per documentation https://docs.pydantic.dev/latest/usage/exporting_models/#modeldict there is no such functionality. I guess you should use just the exclude option or if that doesn't cover your case just remove the key pair afterwards. 

A bit too specific and potentially speed expensive for a feature IMO but wouldn't hurt to mark as a feature request.

**Reply by JonathanLoscalzo** [reply]
I ended up overriding the `dict`method of an own "BaseModel". 
Regarding to the `Field`implementation, I don't have experience working on that.

**Reply by chrisK824** [reply]
Cool 😎
If you could mark the answer as accepted would appreciate it ☺️
